### ISSUE TITLE
Can not edit the cluster when using private registry

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#21711]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug
It is found during testing Rancher:v2.1.11 HA mode in an air-gapped environment, but it should be a general UI bug  

**Steps to reproduce (least amount of steps as possible):**
- run Rancher:v2.1.11
- add a new custom cluster 
- during the creating process, edit the cluster as YAML to add the info of a private registry 
```
private_registries:
- url: **********
  user: **********
  password: **********
  is_default: true
```
- click save 
- the cluster is provisioned successfully and active
- edit the cluster to do any change
- click save

**Result:**
- error message in UI
<img width="1537" alt="Screen Shot 2019-07-19 at 3 34 31 PM" src="https://user-images.githubusercontent.com/6218999/61737299-c36f3900-ad3c-11e9-9d22-7b1a2b019471.png">

**Comment by alena1108** [comment]
@sangeethah if this issue is fixed in 2.2.x, then I'm fine closing it.

**Comment by deniseschannon** [comment]
Fixed in 2.2.x and if someone hits this in 2.1.x, they will need to use the API to edit their cluster.
